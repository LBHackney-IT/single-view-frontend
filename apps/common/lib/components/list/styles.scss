@import "~lbh-frontend/lbh/base";
@import "~lbh-frontend/lbh/core/typography";
