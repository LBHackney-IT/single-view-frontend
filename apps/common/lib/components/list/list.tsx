import React, { forwardRef, isValidElement } from "react";

import cn from "classnames";

import type * as Polymorphic from "@radix-ui/react-polymorphic";
import "./styles.scss";

export interface ListProps {
  variant?: "base" | "bullets" | "numbers";
}

export type ListComponent = Polymorphic.ForwardRefComponent<"ul", ListProps>;

export const List: ListComponent = forwardRef(function List(
  { as: ListComp = "ul", variant = "base", className, children, ...props },
  ref,
) {
  return (
    <ListComp
      ref={ref}
      className={cn(
        "lbh-list",
        {
          "lbh-list--bullet": variant === "bullets",
          "lbh-list--number": variant === "numbers",
        },
        className,
      )}
      {...props}
    >
      {React.Children.map(
        children,
        (child, index) => child && isValidElement(child) && <li key={index}>{child}</li>,
      )}
    </ListComp>
  );
});
