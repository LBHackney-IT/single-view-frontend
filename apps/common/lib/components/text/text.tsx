import React, { forwardRef } from "react";

import cn from "classnames";

import type * as Polymorphic from "@radix-ui/react-polymorphic";
import "./styles.scss";

export interface TextProps {
  variant?: "base" | "regular" | "bold";
  size?: "base" | "xs" | "sm" | "md" | "lg";
}

export type TextComponent = Polymorphic.ForwardRefComponent<"p", TextProps>;

export const Text: TextComponent = forwardRef(function Text(
  { as: TextComp = "p", variant = "base", size = "md", className, ...props },
  ref,
) {
  return (
    <TextComp
      ref={ref}
      className={cn(
        {
          "lbh-body-xs": size === "xs",
          "lbh-body-s": size === "sm",
          "lbh-body-m": size === "md",
          "lbh-body-l": size === "lg",
          "lbh-!-font-weight-regular": variant === "regular",
          "lbh-!-font-weight-bold": variant === "bold",
        },
        className,
      )}
      {...props}
    />
  );
});
