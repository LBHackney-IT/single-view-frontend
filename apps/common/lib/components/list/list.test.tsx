import React from "react";

import { render, testA11y } from "@hackney/mtfh-test-utils";

import { Text } from "../text";
import { List } from "./list";

test("it renders correctly", async () => {
  const { container } = render(
    <List>
      <Text>string1</Text>
      <Text>string2</Text>
      <Text>string3</Text>
    </List>,
  );
  expect(container).toMatchSnapshot();
  await testA11y(container);
});

test("it renders with bullets", async () => {
  const { container } = render(
    <List variant="bullets">
      <Text>string1</Text>
      <Text>string2</Text>
      <Text>string3</Text>
    </List>,
  );
  expect(container).toMatchSnapshot();
});

test("it renders with numbers", async () => {
  const { container } = render(
    <List variant="numbers" as="ol">
      <Text>string1</Text>
      <Text>string2</Text>
      <Text>string3</Text>
    </List>,
  );
  expect(container).toMatchSnapshot();
});
