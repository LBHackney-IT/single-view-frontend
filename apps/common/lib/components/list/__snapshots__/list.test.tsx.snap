// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div>
  <ul
    class="lbh-list"
  >
    <li>
      <p
        class="lbh-body-m"
      >
        string1
      </p>
    </li>
    <li>
      <p
        class="lbh-body-m"
      >
        string2
      </p>
    </li>
    <li>
      <p
        class="lbh-body-m"
      >
        string3
      </p>
    </li>
  </ul>
</div>
`;

exports[`it renders with bullets 1`] = `
<div>
  <ul
    class="lbh-list lbh-list--bullet"
  >
    <li>
      <p
        class="lbh-body-m"
      >
        string1
      </p>
    </li>
    <li>
      <p
        class="lbh-body-m"
      >
        string2
      </p>
    </li>
    <li>
      <p
        class="lbh-body-m"
      >
        string3
      </p>
    </li>
  </ul>
</div>
`;

exports[`it renders with numbers 1`] = `
<div>
  <ol
    class="lbh-list lbh-list--number"
  >
    <li>
      <p
        class="lbh-body-m"
      >
        string1
      </p>
    </li>
    <li>
      <p
        class="lbh-body-m"
      >
        string2
      </p>
    </li>
    <li>
      <p
        class="lbh-body-m"
      >
        string3
      </p>
    </li>
  </ol>
</div>
`;
